src/store/wallet.js

```diff
@@ -46,6 +51,32 @@ function _useWallet(initialWallet = Nothing(), initialMnemonic = Nothing()) {
   const [networkSeed, setNetworkSeed] = useState(Nothing());
   const [networkRevision, setNetworkRevision] = useState(Nothing());
 
+  const [authToken, setAuthToken] = useState(Nothing());
+
+  const { web3 } = useNetwork();
+
+  useEffect(() => {
+    (async () => {
+      if (
+        !Just.hasInstance(wallet) ||
+        !Just.hasInstance(web3) ||
+        NONCUSTODIAL_WALLETS.has(walletType)
```

Comment:
What's with this last check? Doesn't `getAuthToken` support noncustodial wallets just fine?
It's not immediately obvious why we insist on doing `setAuthToken` within the login page code for noncustodial wallets.

Reply:
Because the signing for noncustodial wallets is async and can block indefinitely and we want to prevent login until the signing occurs. To do this we await getAuthToken() inside the Login page for all noncustodial wallets, and generate the signing on login for the others.